cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -479,6 +479,29 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 		}
 	}
 
+	if a.ScaleUpToNodeGroupMinSizeEnabled {
```

Comment:
This whole fucntion is already way too long and the block you're adding is essentially identical to regular scale up one (except for scale up function used). It would make the code easier to maintain if handling metrics & scale up status processor was extracted to a separate function, e.g.: `func (a *StaticAutoscaler) TriggerScaleUp(scaleUp func() (*status.ScaleUpStatus, errors.AutoscalerError)) (*status.ScaleUpStatus, errors.AutoscalerError)` or similar.

Reply:
Sounds good, will do it in the next iteration.